lib/ansible/module_utils/network/eos/eos.py

```diff
@@ -205,7 +205,10 @@ def load_config(self, commands, commit=False, replace=False):
             return conn.load_config(commands, commit, replace)
         except ConnectionError as exc:
             message = getattr(exc, 'err', exc)
-            self._module.fail_json(msg="Error on executing commands %s" % commands, data=to_text(message, errors='surrogate_then_replace'))
+            if 'ConnectionError' in to_text(message):
```

Comment:
Why is this additional check required here?
Since error response from remote host is added as part of data this change is not required imo